File: Medium/Python/1605. Find Valid Matrix Given Row and Column Sums.py
class Solution(object):
    def restoreMatrix(self, rowSum, colSum):
        mat = [[0] * len(colSum) for i in range(len(rowSum))]

        # Fill each spot of the matrix by finding the min of
        # rowSum[i] and colSum[j], then place it into the
        # matrix. Subtract mat[i][j] from rowSum[i]
        # and colSum[j]
        for i in range(len(rowSum)):
            for j in range(len(colSum)):
                mat[i][j] = min(rowSum[i], colSum[j])
                rowSum[i] = rowSum[i] - mat[i][j]
                colSum[j] = colSum[j] - mat[i][j]

        return mat

File: Medium/Java/1605. Find Valid Matrix Given Row and Column Sums.java
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        int[][] mat = new int[rowSum.length][colSum.length];

        // Fill each spot of the matrix by finding the min of
        // rowSum[i] and colSum[j], then place it into the
        // matrix. Subtract mat[i][j] from rowSum[i]
        // and colSum[j]
        for(int i = 0; i < rowSum.length; i++){
            for(int j = 0; j < colSum.length; j++){
                mat[i][j] = Math.min(rowSum[i], colSum[j]);
                rowSum[i] -= mat[i][j];
                colSum[j] -= mat[i][j];
            }
        }

        return mat;
    }
}
